### DISCUSSION TITLE
texlab/latex: long completions move view

### DISCUSSION THREAD

**Discussion by govynnus** [kakoune-lsp/kakoune-lsp#778] [Q&A]
Hello,
First thanks for kakoune-lsp, it's very useful :)

I'm using texlab for latex support, and have run into an annoyance with completions for `\cite`.
I'm not sure if this is a kak-lsp or texlab issue, but asking here first.

Basically, when using <c-n> to choose between completions, all the details of the relevant bibliography entry are temporarily inserted,
which shifts the view way to the right so I can't really see what's going on. Then when the completion is finished the view doesn't return to it's previous state and I have to manually navigate back.

Do you know how I could prevent this from happening? Or how I could get the completions to show up in a different way?

Thanks!
Callum

**Comment by krobelus** [comment]
Hmm texlab passes the entire entry as filterText (which might make sense because it allows you to also filter by title etc) so this is an unfortunate consequence of 25076db4 (Fake filterText support by adding insertText as snippet, 2022-09-24).
I'm not yet sure if we want to change that; removing filterText unconditionally would break other use cases mentioned in that commit.
Maybe we can add some kind of heuristic..

As a workaround you can use `add-highlighter global/my-wrap wrap -word -indent -width %val{window_width} -marker ^`,
then the view port does not change. Of course you'd need to `remove-highlighter global/my-wrap` and add it back whenever the viewport might have changed (`hook g WinResize .* ...` should work)

**Reply by govynnus** [reply]
Thanks! I'll try that and report back.

**Comment by govynnus** [comment]
I've gone with just enabling wrapping for latex files which works well enough for me. Thanks for the tip @krobelus, I didn't think about wrapping!

```
hook global WinSetOption filetype=(latex) %{
    lsp-enable-window
    add-highlighter buffer/ wrap -word -indent -marker '↵ '
}
```
